### ISSUE TITLE
`terraform-ls: cannot execute binary file`

### ISSUE THREAD

**Issue by sarayourfriend** [hashicorp/vscode-terraform#1103]
## Versions

This bug is reproducible in:
 - [x] the **latest** version of the extension (below)
 - [ ] the **latest** version of the language server (below) (Not sure how to answer this one, see workarounds below)

### Extension

```
v2.20.1
```

### Language Server

I'm not sure what version of `terraform-ls` is being shipped with the inspection because the binary is not able to be executed.

### VS Code

```
Version: 1.67.1
Commit: da15b6fd3ef856477bf6f4fb29ba1b7af717770d
Date: 2022-05-10T21:35:59.370Z
Electron: 17.4.3
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Linux x64 5.17.5-arch1-1
```

### Operating System

```
➜  ~ uname -a
Linux saras-laptop 5.17.5-arch1-1 #1 SMP PREEMPT Wed, 27 Apr 2022 20:56:11 +0000 x86_64 GNU/Linux
```

### Terraform Version

```
➜  ~ terraform version
Terraform v1.1.9
on linux_amd64
```

### Steps To Reproduce

`terraform-ls` should be an executable binary.

Note, running `chmod +x` on the binary file did not help.

### Actual Behavior

The binary is not executable.
 
### Additional context

The workaround I'm using currently (which is working great, BTW, amazing work on the extension, the experience with and without it is night and day) is to install `terraform-ls` from the AUR and update my VSCode settings with the following:

```
    "terraform.languageServer": {
        "external": true,
        "pathToBinary": "/usr/bin/terraform-ls",
        "args": [
            "serve"
        ],
        "maxNumberOfProblems": 100,
        "trace.server": "off"
    }
```

**Comment by radeksimko** [comment]
Hi @sarayourfriend 
A few questions so we can try to reproduce the problem.

 - Can you confirm you're running on 64bit Intel-based CPU (amd64/x64)?
 - Can you confirm you installed the official MS build of VS Code from https://code.visualstudio.com/Download
 - Can you confirm you installed the extension from the official [Marketplace](https://marketplace.visualstudio.com/)?

In general VS Code decides what build of the extension (along with the relevant build of LS) to pick. There are some known issues with the last uploaded builds in OpenVSX which some 3rd party VS Code builds likely use. We're trying to resolve that as part of https://github.com/hashicorp/vscode-terraform/pull/1064 but waiting for ovsx maintainers currently.

**Comment by sarayourfriend** [comment]
> Can you confirm you're running on 64bit Intel-based CPU (amd64/x64)?

Yes, an i7 11th gen mobile.

The rest, no I am not, I'm using the version available from the Arch repositories, which I believe is indeed the OSS version of VSCode (no MS binary additions).

Feel free to close this if the issue is already documented with my setup. Thanks for the fast response!

**Comment by radeksimko** [comment]
Understood. It would be still good to know if this is at all related to https://github.com/hashicorp/vscode-terraform/pull/1064 and whether publishing correct packages to OpenVSX will resolve your issues.

The version you mentioned (`2.20.1`) would suggest it is - as that is the latest extension version we published there (unlike [2.22.0 on the official MS Marketplace](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform)) and I know for a fact the [package uploaded for that version is unfortunately broken](https://github.com/hashicorp/vscode-terraform/pull/1064#pullrequestreview-960905567).

If you agree, we can close this as a duplicate of https://github.com/hashicorp/vscode-terraform/pull/1064

**Comment by sarayourfriend** [comment]
Ah, okay! I'm fine closing this issue. I will keep an eye out for updates to the extension in OpenVSX and if it isn't fixed on the next release there then I'll make a not here about it :grin:

**Comment by radeksimko** [comment]
FWIW in the meantime you may be able to downgrade to [2.19.0](https://open-vsx.org/extension/hashicorp/terraform/2.19.0) which was the last version built as universal (rather than platform specific) - but of course managing LS on your own like you mentioned is also a good workaround.

As mentioned in the linked issue we are currently blocked by https://github.com/eclipse/openvsx/issues/450

**Comment by tagirb** [comment]
As a workaround one can download the platform-specific `terraform-ls` binary archive from [hashicorp/terraform-ls release page](https://github.com/hashicorp/terraform-ls/releases/latest) and replace the broken binary in the VS Code extensions directory (on my Linux it is `~/.vscode-oss/extensions/hashicorp.terraform-2.20.1/bin`).
